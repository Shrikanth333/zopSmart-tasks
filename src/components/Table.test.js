import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import SimpleTables from './Table';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'
import toJSON from 'enzyme-to-json'
configure({ adapter: new Adapter() });
const mockStore = configureMockStore();


  




describe("Table Component should render without any error", () => {
    let store;
    let wrapped;
    let props
beforeEach(()=>{
     props ={
        numberOfRows:4,
        numberOfColumns:3,
     rowsData:[[0,0,0,0,0]],
     
        //  fetchAllRows: jest.fn(),
     }
    store = mockStore({rowsData:[[0,0,0,0,0]]});
    store.dispatch = jest.fn();
    wrapped =mount(
        <Provider store={store}>
            <SimpleTables  {...props}/>
         </Provider>
    )
})
// const myContainer=wrapped.dive()

    it("Table ", () => {
        expect(wrapped).toMatchSnapshot();
    });
});
