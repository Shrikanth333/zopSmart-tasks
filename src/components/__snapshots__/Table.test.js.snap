// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component should render without any error Table  1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [MockFunction],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <SimpleTables
    numberOfColumns={3}
    numberOfRows={4}
    rowsData={
      Array [
        Array [
          0,
          0,
          0,
          0,
          0,
        ],
      ]
    }
  />
</ContextProvider>
`;
